utils/subfiling_vfd/h5fuse.sh.in

```diff
@@ -104,62 +149,125 @@ if test -z "$subfile_dir"; then
     exit $FAILED
 fi
 
-subfiles=( $( sed -e '1,/subfile_dir=/d' "$file_config" ) )
-#for i in "${subfiles[@]}"; do
-#      echo "$i"
-#done
+# For bash 4.4+
+mapfile -t subfiles < <( sed -e '1,/subfile_dir=/d' "$file_config" )
 if [ ${#subfiles[@]} -eq 0 ]; then
     echo -e "$RED failed to find subfiles list in $file_config $NC"
     exit $FAILED
 fi
+nsubfiles=${#subfiles[@]}
+
+# Get the number of local subfiles
+subfiles_loc=()
+subfiles_size=()
+for i in "${subfiles[@]}"; do
+    subfile="${subfile_dir}/${i}"
+    if [ -f "${subfile}" ]; then
+        subfiles_loc+=("$subfile")
+        subfiles_size+=($(wc -c "${subfile}" | awk '{print $1}'))
+    else
+        subfiles_size+=(0)
+    fi
+done
+
+START="$(date +%s%N)"
+
+mpi_rank=0
+mpi_size=1
+nstart=1
+nend=$nsubfiles
+
+if [ "$parallel" == "true" ]; then
+
+    hex=$(hexdump -n 16 -v -e '/1 "%02X"' /dev/urandom)
+    c_exec="h5fuse_"${hex}
+    c_src=${c_exec}.c
+
+    # Generate and compile an MPI program to get MPI rank and size
+    if [ ! -f "${c_src}" ]; then
+        gen_mpi
+        CC=@CC@
+        ${CC} "${c_src}" -o "${c_exec}"
```

Comment:
This script gets submitted as a batch job right? I wonder if there are cases where mpicc might not exist on the compute node and program compilation may fail here.

Reply:
I've not come across a case where the compiler is not available on the computer nodes since facilities encourage codes to be compiled as a batch job.